Sources/RegexBuilder/CharacterClass.swift

```diff
@@ -0,0 +1,209 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import _RegexParser
+@_spi(RegexBuilder) import _StringProcessing
+
+public struct CharacterClass {
+  internal var ccc: DSLTree.CustomCharacterClass
+  
+  init(_ ccc: DSLTree.CustomCharacterClass) {
+    self.ccc = ccc
+  }
+  
+  init(unconverted model: _CharacterClassModel) {
+    // FIXME: Implement in DSLTree instead of wrapping an AST atom
+    switch model.makeAST() {
+    case .atom(let atom):
+      self.ccc = .init(members: [.atom(.unconverted(atom))])
+    default:
+      fatalError("Unsupported _CharacterClassModel")
+    }
+  }
+  
+  init(property: AST.Atom.CharacterProperty) {
+    // FIXME: Implement in DSLTree instead of wrapping an AST atom
+    let astAtom = AST.Atom(.property(property), .fake)
+    self.ccc = .init(members: [.atom(.unconverted(astAtom))])
+  }
+}
+
+extension CharacterClass: RegexComponent {
+  public var regex: Regex<Substring> {
+    return Regex(node: DSLTree.Node.customCharacterClass(ccc))
+  }
+}
+
+extension CharacterClass {
+  public var inverted: CharacterClass {
+    CharacterClass(ccc.inverted)
+  }
+}
+
+extension RegexComponent where Self == CharacterClass {
+  public static var any: CharacterClass {
+    .init(DSLTree.CustomCharacterClass(members: [.atom(.any)]))
+  }
+
+  public static var anyGrapheme: CharacterClass {
+    .init(unconverted: .anyGrapheme)
+  }
+
+  public static var whitespace: CharacterClass {
+    .init(unconverted: .whitespace)
+  }
+  
+  public static var digit: CharacterClass {
+    .init(unconverted: .digit)
+  }
+  
+  public static var hexDigit: CharacterClass {
+    .init(DSLTree.CustomCharacterClass(members: [
+      .range(.char("A"), .char("F")),
+      .range(.char("a"), .char("f")),
+      .range(.char("0"), .char("9")),
+    ]))
+  }
+
+  public static var horizontalWhitespace: CharacterClass {
+    .init(unconverted: .horizontalWhitespace)
+  }
+
+  public static var newlineSequence: CharacterClass {
+    .init(unconverted: .newlineSequence)
+  }
+
+  public static var verticalWhitespace: CharacterClass {
+    .init(unconverted: .verticalWhitespace)
+  }
+
+  public static var word: CharacterClass {
+    .init(unconverted: .word)
+  }
+}
+
+extension RegexComponent where Self == CharacterClass {
+  /// Returns a character class that matches any character in the given string
+  /// or sequence.
+  public static func anyOf<S: Sequence>(_ s: S) -> CharacterClass
```

Comment:
Should the name align with `ChoiceOf`?  Could this be confusing with the `ChoiceOf` type, or perhaps be folded as an initializer under `ChoiceOf` so that the developer won't need to write `CharacterClass.anyOf`?

Reply:
I had and removed a `ChoiceOf` initializer, since one thing that's distinct about character classes vs. other regex components is that character classes support inversion. Maybe just a `CharacterClass` initializer instead of this, since you often need the type name anyway?

Reply:
Good point... An initializer sounds good.